### PR TITLE
EPMLSTRCMW-280 Move buildConfiguration from ProjectFileService to ProjectFileConfiguration

### PR DESCRIPTION
**Pull request by Nikolai-Isaev**
refactor: Move buildConfiguration from ProjectFileService to ProjectFileConfiguration and corresponding controllers

### CODE REVIEW

File: services/src/test/scala/cromwell/pipeline/service/ProjectConfigurationServiceTest.scala [hunk 1]
```diff
@@ -159,5 +171,51 @@ class ProjectConfigurationServiceTest extends AsyncWordSpec with Matchers with M
...
+          .buildConfiguration(projectId, projectFile.path, optionVersion, userId)
+          .map(
+            builtConfiguration =>
+              builtConfiguration shouldBe ProjectConfiguration(
+                id = builtConfiguration.id,
+                projectId = projectId,
+                active = true,
+                projectFileConfigurations = List(ProjectFileConfiguration(projectFile.path, Nil)),
+                version = ProjectConfigurationVersion.defaultVersion
+              )
```
Comment (DruzhininPavel) [inline services/src/test/scala/cromwell/pipeline/service/ProjectConfigurationServiceTest.scala:194]: I think you should exclude configuration from test and use created object.

**Review by DruzhininPavel — CHANGES_REQUESTED** [review]
Looks good, but one moment is..

File: services/src/test/scala/cromwell/pipeline/service/ProjectConfigurationServiceTest.scala [hunk 2]
```diff
@@ -159,5 +171,51 @@ class ProjectConfigurationServiceTest extends AsyncWordSpec with Matchers with M
...
+        when(projectService.getUserProjectById(projectId, userId)).thenReturn(Future.successful(dummyProject))
+        when(projectVersioning.getFile(dummyProject, projectFile.path, optionVersion))
+          .thenReturn(Future.successful(Right(projectFile)))
+        when(womTool.inputsToList(projectFileContent.content)).thenReturn(Right(Nil))
+
+        configurationService
+          .buildConfiguration(projectId, projectFile.path, optionVersion, userId)
+          .map(
+            builtConfiguration =>
+              builtConfiguration shouldBe ProjectConfiguration(
```
Comment (gominasian) [inline services/src/test/scala/cromwell/pipeline/service/ProjectConfigurationServiceTest.scala:188]: As for me it's a little bit confusing buildConfiguration / builtConfiguration. Should we use another name? Or just underscore?

**Review by gominasian — APPROVED** [review]
Good job!

File: services/src/test/scala/cromwell/pipeline/service/ProjectConfigurationServiceTest.scala (hunk 2 above)
Comment (Nikolai-Isaev) [reply services/src/test/scala/cromwell/pipeline/service/ProjectConfigurationServiceTest.scala:188]: Thanks, that's indeed confusing. My task was just to move this code. I would prefer not to change it or change as little as possible

File: services/src/test/scala/cromwell/pipeline/service/ProjectConfigurationServiceTest.scala (hunk 1 above)
Comment (Nikolai-Isaev) [reply services/src/test/scala/cromwell/pipeline/service/ProjectConfigurationServiceTest.scala:194]: Can you be more specific? If you mean to use dummyConf, how to compare it by random Id?

**Comment by Nikolai-Isaev** [comment]
Update: `buildConfiguration` added into `ProjectConfigurationServiceTestImpl`

File: services/src/test/scala/cromwell/pipeline/service/ProjectConfigurationServiceTest.scala (hunk 1 above)
Comment (DruzhininPavel) [reply services/src/test/scala/cromwell/pipeline/service/ProjectConfigurationServiceTest.scala:194]: Ok, in this case used id from builtConguration, maybe I was wrong and you shouldn`t move this pice of code. Sorry.

**Review by DruzhininPavel — APPROVED** [review]
No more questions.

File: controllers/src/main/scala/cromwell/pipeline/controller/ProjectConfigurationController.scala [hunk 3]
```diff
@@ -48,8 +49,28 @@ class ProjectConfigurationController(projectConfigurationService: ProjectConfigu
     }
   }
 
+  private def buildConfiguration(projectId: ProjectId)(implicit accessToken: AccessTokenContent): Route = get {
+    path("file" / Path) { projectFilePath =>
```
Comment (myazinn) [inline controllers/src/main/scala/cromwell/pipeline/controller/ProjectConfigurationController.scala]:
```suggestion
    path("files" / Path) { projectFilePath =>
```

**Review by myazinn — COMMENTED** [review]
Only one minor comment from my side

File: controllers/src/main/scala/cromwell/pipeline/controller/ProjectConfigurationController.scala (hunk 3 above)
Comment (Nikolai-Isaev) [reply controllers/src/main/scala/cromwell/pipeline/controller/ProjectConfigurationController.scala]: Fixed

**Review by myazinn — APPROVED** [review]

**Review by Dmitrii-Shevchenko — APPROVED** [review]

**Pull request merged**